src/org/armedbear/lisp/ZipCache.java

```diff
@@ -423,18 +422,79 @@ synchronized public static Archive getArchive(JarPathname jar) {
     if (innerJars.equals(NIL)) {
       return getArchiveFile(jar);
     } else {
+      result = getArchiveStreamFromFile(jar);
+      cache.put(result.root, result); 
+
+      JarPathname nextArchive = new JarPathname();
+      nextArchive
+        .setDevice(new Cons(rootJar,
+                            new Cons(innerJars.car(), NIL)))
+        .setDirectory(NIL)
+        .setName(NIL)
+        .setType(NIL);
+      
+      innerJars = innerJars.cdr();
+      while (innerJars.car() != NIL) {
+        Pathname nextJarArchive = (Pathname)innerJars.car();
+
+        JarPathname nextAsEntry = new JarPathname();
+        nextAsEntry
+          .setDevice(nextArchive.getDevice())
+          .setDirectory(nextJarArchive.getDirectory())
+          .setName(nextJarArchive.getName())
+          .setType(nextJarArchive.getType());
+        // FIXME 
+        // The pathnames for subsquent entries in a PATHNAME-JAR are relative.  Should they be?
+        if (nextAsEntry.getDirectory().car().equals(Keyword.RELATIVE)) {
+          LispObject directories = nextAsEntry.getDirectory();
```

Comment:
One whole line for the `.cdr` is sprawly.  Why not 

    LispObject directories = nextAsEntry.getDirectory().cdr();

?